### PR TITLE
Enhance Y-Axis Scaling on Net Worth Graph

### PR DESCRIPTION
**Pull request by Crazypkr1099**

### CODE REVIEW

**Comment by Crazypkr1099** [comment]
The enhancement adds for scaling that makes sense based on your Net-Worth. Currently, if your Net-Worth is high but your income up/down is low, its very hard to see on the chart. I added code to specify what your y min-max should be so it's easy to see the increase/decrease.

Below is an examples:

**BEFORE ENHANCEMENT**
Networth-before-July-Sept

**AFTER ENHANCEMENT**
Networth-after-July-Sept

**ANOTHER EXAMPLE (LEFT = Enhanced)**
Networth-after-may-Sept

File: packages/desktop-client/src/components/reports/graphs/net-worth-spreadsheet.tsx [hunk 1]
```diff
@@ -152,5 +163,7 @@ function recalculate(data, start, end) {
     },
     netWorth: endNetWorth,
     totalChange: endNetWorth - startNetWorth,
+    lowestNetWorth: lowestNetWorth,
+    highestNetWorth: highestNetWorth,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/graphs/net-worth-spreadsheet.tsx:167]: **🥜 nitpick:** ‏
```suggestion
    lowestNetWorth,
    highestNetWorth,
```

File: packages/desktop-client/src/components/reports/graphs/net-worth-spreadsheet.tsx [hunk 2]
```diff
@@ -140,6 +142,15 @@ function recalculate(data, start, end) {
...
 
     arr.push({ x, y: integerToAmount(total), premadeLabel: label });
+
+    arr.forEach(item => {
+      if (item.y < lowestNetWorth || lowestNetWorth == null) {
+        lowestNetWorth = item.y;
+      }
+      if (item.y > highestNetWorth || highestNetWorth == null) {
+        highestNetWorth = item.y;
+      }
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/graphs/net-worth-spreadsheet.tsx:152]: **🥜 nitpick:** ‏strict null checks
```suggestion
      if (item.y < lowestNetWorth || lowestNetWorth === null) {
        lowestNetWorth = item.y;
      }
      if (item.y > highestNetWorth || highestNetWorth === null) {
        highestNetWorth = item.y;
      }
```

File: packages/desktop-client/src/components/reports/graphs/NetWorthGraph.tsx [hunk 3]
```diff
@@ -38,6 +44,7 @@ function NetWorthGraph({ style, graphData, compact }: NetWorthGraphProps) {
             scale={{ x: 'time', y: 'linear' }}
             theme={chartTheme}
             domainPadding={{ x: 0, y: 10 }}
+            domain={domainMinMax}
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/graphs/NetWorthGraph.tsx:47]: **💬 suggestion:** ‏ideally we use the same variable name here. Aliasing and renaming things down multiple layers of code leads to difficult to read code.

File: packages/desktop-client/src/components/reports/graphs/NetWorthGraph.tsx [hunk 4]
```diff
@@ -21,8 +21,14 @@ type NetWorthGraphProps = {
   style?: CSSProperties;
   graphData;
   compact: boolean;
+  domainMinMax;
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/graphs/NetWorthGraph.tsx:24]: **💬 suggestion:** ‏could you please add a type for the new prop?

File: packages/desktop-client/src/components/reports/NetWorth.js [hunk 5]
```diff
@@ -133,6 +132,12 @@ export default function NetWorth() {
           start={start}
           end={end}
           graphData={data.graphData}
+          domainMinMax={{
+            y: [
+              data.lowestNetWorth - data.lowestNetWorth * 0.01,
+              data.highestNetWorth + data.highestNetWorth * 0.01,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/NetWorth.js:138]: **💬 suggestion:** ‏could be simplified
```suggestion
              data.lowestNetWorth * 0.99,
              data.highestNetWorth * 1.01,
```

**Review by MatissJanis — CHANGES_REQUESTED** [review]
Really cool addition! This makes the net-worth graph so much more usable.

Thanks!

File: packages/desktop-client/src/components/reports/graphs/NetWorthGraph.tsx [hunk 6]
```diff
@@ -17,12 +17,20 @@ import Tooltip from '../Tooltip';
 
 import { Area } from './common';
 
+type domainProp = [number, number];
+
 type NetWorthGraphProps = {
   style?: CSSProperties;
   graphData;
   compact: boolean;
+  domain: domainProp;
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/graphs/NetWorthGraph.tsx:26]: **🔨 warning:** ‏`[number, number]` won't be correct here. You could either do `{ y?: [number, number]; x?: [number, number] }` or..

```suggestion
  domain: React.ComponentProps<typeof Chart>["domain"];
```

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**